### ISSUE TITLE
API v2: genre and language fields missing from beatmaps responses

### ISSUE THREAD

**Issue by MrHeliX** [ppy/osu-web#10726]
In the deprecated osu! API, using the `/api/get_beatmaps` endpoint, you will receive, among others, the following fields:
```
{
    "genre_id": "2", // 0 = any, 1 = unspecified, 2 = video game, 3 = anime, 4 = rock, 5 = pop, 6 = other, 7 = novelty, 9 = hip hop, 10 = electronic, 11 = metal, 12 = classical, 13 = folk, 14 = jazz (note that there's no 8)
    "language_id": "5", // 0 = any, 1 = unspecified, 2 = english, 3 = japanese, 4 = chinese, 5 = instrumental, 6 = korean, 7 = french, 8 = german, 9 = swedish, 10 = spanish, 11 = italian, 12 = russian, 13 = polish, 14 = other
}
```

In the v2 API, these fields are missing from both the Beatmap(set) and Beatmap(set)Extended objects. I would like to see these fields added to the v2 API responses as well

**Comment by notbakaneko** [comment]
`genre` and `language` are part of the detailed [GET api/v2/beatmapsets/{beatmapset}](https://osu.ppy.sh/docs/index.html#get-apiv2beatmapsetsbeatmapset) response, they're not included in listing responses

**Comment by MrHeliX** [comment]
Yeah I'm aware they're not included, that's why I created the issue :)

**Comment by SoulDee** [comment]
> `genre` and `language` are part of the detailed [GET api/v2/beatmapsets/{beatmapset}](https://osu.ppy.sh/docs/index.html#get-apiv2beatmapsetsbeatmapset) response, they're not included in listing responses

I encountered the same problem. I am building a mirror resource website, which requires fields such as Genre.id/ Language.id as filtering conditions, and I need to obtain a large amount of data at one time when initializing the database.

By looking at the documentation and source code of the v2 api, I found that it seems that the two APIs "api/v2/beatmapsets/lookup" and "api/v2/beatmapsets/{beatmapset}" can return detailed Beatmapset information, but only support one request at a time.

"/beatmapsets/search" supports returning 50 at a time, but the returned data is incomplete.

From the total field returned by "/beatmapsets/search", I see that there are more than one million beatmapsets at present. If you can only request one data at a time by id, this will generate a very large number of requests, and even at a request rate of 1,200 times per minute, it will still take more than ten hours of continuous requests.

Is there any way to quickly obtain more Beatmapset details and data at one time?

**Comment by peppy** [comment]
Seems like these could be added with no overhead (apart from slightly increased response size)?

**Comment by nanaya** [comment]
added in #12167
